### PR TITLE
[CRT_APITEST] Add testcase for mbtowc()

### PR DESCRIPTION
**Pull request by Fraizeraust**
**Windows XP**
Capture2
**Windows Server 2003**

**ReactOS (with #2563 applied to build)**
Capture

### CODE REVIEW

File: modules/rostests/apitests/crt/mbtowc.c [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#include <stdio.h>
+#include <stdlib.h>
+
+#define ok_wchar(x, y) ok_int(x, y)
+
+START_TEST(mbtowc)
+{
+    size_t Length;
+    wchar_t BufferDest[3];
+    char ch = 'A';
```
Comment (ThFabba) [inline modules/rostests/apitests/crt/mbtowc.c:20]: A possible future improvement would be to allocate a buffer for this using `AllocateGuarded`, to show that only the first character is ever accessed.

File: modules/rostests/apitests/crt/mbtowc.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    /* The multibyte argument is invalid */
+    Length = mbtowc(BufferDest, NULL, 0);
+    ok(Length == 0, "Expected 0 characters to be converted but got %u.\n", Length);
+
+    /* The multibyte argument is invalid but count number for examination is correct */
+    Length = mbtowc(BufferDest, NULL, MB_CUR_MAX);
+    ok(Length == 0, "Expected 0 characters to be converted but got %u.\n", Length);
+
+    /* Convert the character and validate the output that we should get */
+    Length = mbtowc(BufferDest, &ch, MB_CUR_MAX);
```
Comment (ThFabba) [inline modules/rostests/apitests/crt/mbtowc.c:43]: Another useful test would be this with a NULL wchar parameter, to show that you can get the length without receiving output.

**Comment by Fraizeraust** [comment]
Both MSVCRT and NTDLL apitest testcases work flawlessly for this function testcase in Windows Server 2003 and ReactOS, although with the exception of NTDLL testcase unit in Windows XP as `ntdll_crt_apitest.exe` prematurely fails with a missing entry point ` _vscwprintf` in ntdll.dll which by default doesn't exist in XP.
